freqtrade/rpc/telegram.py

```diff
@@ -235,17 +235,31 @@ def _format_buy_msg(self, msg: Dict[str, Any]) -> str:
 
         if msg['type'] == RPCMessageType.BUY_FILL:
             message += f"*Open Rate:* `{msg['open_rate']:.8f}`\n"
+            total = msg['amount'] * msg['open_rate']
 
         elif msg['type'] == RPCMessageType.BUY:
             message += f"*Open Rate:* `{msg['limit']:.8f}`\n"\
                        f"*Current Rate:* `{msg['current_rate']:.8f}`\n"
-
-        message += f"*Total:* `({round_coin_value(msg['stake_amount'], msg['stake_currency'])}"
+            total = msg['amount'] * msg['limit']
+        if self._rpc._fiat_converter:
+            total_fiat = self._rpc._fiat_converter.convert_amount(
+                total, msg['stake_currency'], msg['fiat_currency'])
+        else:
+            total_fiat = 0
+        message += f"*Total:* `({round_coin_value(total, msg['stake_currency'])}"
 
         if msg.get('fiat_currency', None):
-            message += f", {round_coin_value(msg['stake_amount_fiat'], msg['fiat_currency'])}"
+            message += f", {round_coin_value(total_fiat, msg['fiat_currency'])}"
 
         message += ")`"
+        if msg.get('sub_trade'):
+            bal = round_coin_value(msg['stake_amount'], msg['stake_currency'])
+            message += f"\n*Balance:* `({bal}"
```

Comment:
I don't think the wording of "balance" is correct in this context.
What i assume you mean is "Remaining" (i sold 100XXX, and still own 150 XXX)?

Reply:
As we used the profit to average the price of previous buy, realized profit can't be calculated for subsequent sells.

For example,
1   buy 100 @ 11
2   sell 50 @ 8
3   sell 50 @ 16

What is the realized and unrealized profit after order 2?
What is the realized profit after order 3?

The final profit is 94.25

Reply:
As we used the profit to average the price of previous buy, realized profit can't be calculated for subsequent sells.

For example without fees,
1   buy 2 @ 10
2   buy 2 @ 5
3  sell 2 @ 6
4   sell 2 @ 9

What is the realized and unrealized profit after order 2 and 3?
What is the realized profit after order 4?

The final profit is 0

Reply:
For example without fees,
1   buy 2 @ 10
2   buy 2 @ 5
3  sell 2 @ 6
4   sell 2 @ 9

What is the realized and unrealized profit after order 2 and 3?
What is the realized profit after order 4?

The final profit is 0

Reply:
well after order 2 it's easy - realized profit is 0 (you didn't sell anything - so i don't think this should even be shown) and unrealized is whatever the result is (something negative, i suppose).

after 3 - your maximum position was 30$ - you sold half of it for a total of 18$ (your average entry price was 7.5$ - your sell was at 6$/pce - so your profit will be ~-3$ (and whatever this results to in relative terms) <- this is the realized profit
You'll still have some unrealized profit - which will be the remaining position vs. current price.

after 4 - realized profit will be "whatever the outcome" (so 0 in this case) - and there's no unrealized profit (you're no longer in a position).

I think you're too focused on this one usecase though (DCA with full lotsizes only) - what for the following scenario (increasing position size because it's going "our way")

1 buy 5 @ 10$
2 buy 5 @ 12$
3 sell 2 @ 15$
4 buy 5 @ 13$
5 sell remaining 13 @17$

After order 3 - you'll have some realized/materialized profit (you've taken profit - so you MUST keep track of how much you earned in this trade already).
You then rebuy on a small pullback - before selling for a substancial gain.

Reply:
1 buy 5 @ 28$
2 buy 5 @ 10$
3 sell 2 @ 11$
4 buy 5 @ 13$
5 sell remaining 13 @17$

Here after order 3, I calculated profit from last order price($10) instead of avg price($12). So the profit is $2 instead of -$2 and subtracted it from the stake amount of the previous orders to reduce the avg price. If like order 2 and 3 happens multiples times, the stake amount will be reduced. Also it reduces the chance of hitting stoploss as we used the profit instead of just tracking.

Reply:
Used quote amount as 1 for the sake of example.

1 buy 1 @ $10
2 buy 1 @ $5
3 sell 1 @ $8



Here, I calculated profit from last order price($5) instead of avg price($7.5). So the profit is $3 instead of $.5 and subtracted it from the stake amount of the previous orders and calculated new avg price.

So the new avg price is $7 instead of $7.5.

If the stoploss sets for -6% after 1 hour, and the price comes to $7 , the trade will be saved by the new avg price.



4 buy 1 @ 7.5
5 sell 1 @ 8.5
And price becomes 4.
stoploss is at -60%

If like order 4 and 5 happens 10 times, there would be no loss. 
If we don't subtract,

Reply:
> your sell was at 6$/pce - so your profit will be ~-3

Here I considered the profit is $2.
In user perspective, we are selling it in profit as we bought at lower price previously.

Reply:
> I'd like to see 2 things in the notification
> 
> * Realized profit
> * unrealized / still open profit
> 
> Both of the above would also make sense in the `/status` context / message.

So at every partial sell, the profit includes the previous profits. As we subtract the profit from stake amount, the realized profit reduces the unrealized profit.

Same happens if we rebuy at higher prices.

Reply:
> your sell was at 6$/pce - so your profit will be ~-3

Here I considered the profit is $2.
In user perspective, we are selling it in profit as we bought at lower price previously.

Reply:
> I'd like to see 2 things in the notification
> 
> * Realized profit
> * unrealized / still open profit
> 
> Both of the above would also make sense in the `/status` context / message.

So at every partial sell, the profit includes the previous profits. As we subtract the profit from stake amount, the realized profit reduces the unrealized profit.

Same happens if we rebuy at higher prices and selling in different lot sizes.


This way it will be easier for strategies to process the trades else the strategy will need to process the realized profits and adjusts stoplosses according to that.

Reply:
Maybe we can sort the orders list by prices

Reply:
> Used quote amount as 1 for the sake of example.

By simplying stake to 1 - you're ignoring the complexity of different stake sizes. While equal size is a possibility we should be prepared for, please assume this as an exceptional case - not "what we base our calculations on".


> Here, I calculated profit from last order price($5) instead of avg price($7.5). So the profit is $3 instead of $.5 and subtracted it from the stake amount of the previous orders and calculated new avg price.

I don't think this is correct behaviour, and this will fail once you use different stake sizes per order (how would this behave if the "last" order is smaller than the amount sold?).

You should instead use the (weighted) average price of all orders so far - obviously subtracting already sold amounts (so the average open price may change as more amount is sold).

> Maybe we can sort the orders list by prices

Orders should be sorted by date - never by price.


> So at every partial sell, the profit includes the previous profits. As we subtract the profit from stake amount, the realized profit reduces the unrealized profit.

It's a point of discussion if the regular profit column should contain all profit (realized + unrealized) - and one i would leave up to you for now (assuming it's properly outlined in the documentation).

I'd however still want to see "realized" profit as a separate datapoint  - as this corresponds to "locked in" profit - which is an important point (if this trade goes completely against me - how much profit did i already lock in).

Reply:
> how would this behave if the "last" order is smaller than the amount sold?

It will take the previous order price for the remaining amount.

Reply:
> so your profit will be ~-3$

Shouldn't it be subtracted from the stake amount?
So that the ROI and stoploss will be adjusted.

Reply:
Done